server.js

```diff
@@ -1,10 +1,32 @@
 const express = require('express');
+const socket = require('socket.io');
+
+const port = 8000;
+const tasks = [];
 
 const app = express();
-const server = app.listen(process.env.PORT || 8000, () => {
-  console.log('Server is running...');
+
+const server = app.listen(port, () => {
+    console.log(`App running at http://localhost:${port}`);
+});
+
+const io = socket(server);
+
+io.on('connection', (socket) => {
+    console.log('New client! Its id – ' + socket.id);
+    socket.emit('updateData', tasks);
+
+    socket.on('addTask', (task) => {
+        tasks.push(task);
+        socket.broadcast.emit('addTask', task);
```

Comment:
Pytanie - czy jak broadcastujesz, to czy dana wiadomość jest również wysyłana do usera, który przysłał ten task? Jeżeli tak, to czy nie duplikuje Ci tasków po stronie frontu? To samo dla removeTask - do sprawdzenia :)